### PR TITLE
Rails 5.0 upgrade (#2726)

### PR DESCRIPTION
**Pull request by jywarren**
Attempting a PR from #2726 to `master`

### CODE REVIEW

**Comment by jywarren** [comment]
@Souravirus can you try rebasing this? I'm sorry, it looks like the previous #2726 was based against the `rails5.0_upgrade` only - not `master` -- I hope this won't be too much extra; you've rebased at least once already!

**Comment by Souravirus** [comment]
yeah I will do @jywarren

**Comment by Souravirus** [comment]
@jywarren rebasing done and all tests clear(except codeclimate). So, are we ready to merge?

**Comment by jywarren** [comment]
Hmm, that last `push` travis run is stalled maybe? Looking...

**Comment by jywarren** [comment]
restarted it!

**Comment by Souravirus** [comment]
yeah ok

**Comment by Souravirus** [comment]
finally, that pr test is over

**Comment by jywarren** [comment]
OK, it finished! Everything's a bit slow for some reason... danger hasn't updated. Any idea why? Did that last change you made get applied from last time we saw this error?

**Comment by Souravirus** [comment]
No idea why danged didn't updated

**Comment by Souravirus** [comment]
Should we try closing and reopening the pr again

**Comment by jywarren** [comment]
I wonder if `danger` timed out when the Travis tests took so long?

**Comment by Souravirus** [comment]
Yeah might be

**Comment by jywarren** [comment]
yeah i guess... frustrating! :-)

**Comment by Souravirus** [comment]
finally all successful checks

**Comment by jywarren** [comment]
Haha! Fantastic!!!!

**Comment by jywarren** [comment]
Since we rebased, i'm going to give it a quick skim through one more time to be sure. Was the rebase bad?

**Comment by Souravirus** [comment]
Yeah there were some conflicts which I have to clear.

File: config/initializers/wrap_parameters.rb [hunk 1]
```diff
@@ -5,10 +5,10 @@
...
 # Enable parameter wrapping for JSON. You can disable this by setting :format to an empty array.
 ActiveSupport.on_load(:action_controller) do
-  wrap_parameters format: [:json] if respond_to?(:wrap_parameters)
+    wrap_parameters format: [:json]
 end
 
 # To enable root element in JSON for ActiveRecord objects.
 ActiveSupport.on_load(:active_record) do
-  self.include_root_in_json = false
+    self.include_root_in_json = true
```
Comment (jywarren) [inline config/initializers/wrap_parameters.rb]: Aha! I think this may have been responsible for our JSON error. But no need to change anything!

Comment (Souravirus) [reply config/initializers/wrap_parameters.rb]: Yeah correct @jywarren :smile:

**Pull request merged**